webapp/src/app/components/leaderboard/leaderboard.component.ts

```diff
@@ -0,0 +1,153 @@
+    name: 'A',
+    score: 10,
+    changesRequested: 1,
+    approvals: 0,
+    comments: 0
+  },
+  {
+    githubName: 'shadcn',
+    avatarUrl: 'https://avatars.githubusercontent.com/u/124599?v=4',
+    type: LeaderboardEntry.TypeEnum.User,
+    name: 'B',
+    score: 20,
+    changesRequested: 0,
+    approvals: 1,
+    comments: 0
+  },
+  {
+    githubName: 'shadcn',
+    avatarUrl: 'https://avatars.githubusercontent.com/u/124599?v=4',
+    type: LeaderboardEntry.TypeEnum.User,
+    name: 'C',
+    score: 30,
+    changesRequested: 0,
+    approvals: 0,
+    comments: 1
+  },
+  {
+    githubName: 'shadcn',
+    avatarUrl: 'https://avatars.githubusercontent.com/u/124599?v=4',
+    type: LeaderboardEntry.TypeEnum.User,
+    name: 'D',
+    score: 40,
+    changesRequested: 0,
+    approvals: 0,
+    comments: 0
+  },
+  {
+    githubName: 'shadcn',
+    avatarUrl: 'https://avatars.githubusercontent.com/u/124599?v=4',
+    type: LeaderboardEntry.TypeEnum.User,
+    name: 'E',
+    score: 50,
+    changesRequested: 0,
+    approvals: 0,
+    comments: 0
+  },
+  {
+    githubName: 'shadcn',
+    avatarUrl: 'https://avatars.githubusercontent.com/u/124599?v=4',
+    type: LeaderboardEntry.TypeEnum.User,
+    name: 'F',
+    score: 60,
+    changesRequested: 0,
+    approvals: 0,
+    comments: 0
+  },
+  {
+    githubName: 'shadcn',
+    avatarUrl: 'https://avatars.githubusercontent.com/u/124599?v=4',
+    type: LeaderboardEntry.TypeEnum.User,
+    name: 'G',
+    score: 70,
+    changesRequested: 0,
+    approvals: 0,
+    comments: 0
+  },
+  {
+    githubName: 'shadcn',
+    avatarUrl: 'https://avatars.githubusercontent.com/u/124599?v=4',
+    type: LeaderboardEntry.TypeEnum.User,
+    name: 'H',
+    score: 80,
+    changesRequested: 0,
+    approvals: 0,
+    comments: 0
+  }
+];
+
+const sortByScore = (a: LeaderboardEntry, b: LeaderboardEntry) => {
+  if (!b.score) {
+    return -1;
+  }
+  if (!a.score) {
+    return 1;
+  }
+  return b.score - a.score;
+};
+
+@Component({
+  selector: 'app-leaderboard',
+  standalone: true,
+  imports: [
+    TableComponent,
+    TableBodyDirective,
+    TableCaptionDirective,
+    TableCellDirective,
+    TableFooterDirective,
+    TableHeaderDirective,
+    TableHeadDirective,
+    TableRowDirective,
+    PullRequestChangesRequestedIconComponent,
+    PullRequestApprovedIconComponent,
+    PullRequestCommentIconComponent
+  ],
+  templateUrl: './leaderboard.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class LeaderboardComponent {
+  leaderboardService = inject(LeaderboardService);
+
+  query = injectQuery(() => ({
+    queryKey: ['leaderboard'],
+    queryFn: async () => lastValueFrom(this.leaderboardService.getLeaderboard()),
+    gcTime: Infinity
+  }));
+
+  leaderboard = computed(() => {
+    let data = this.query.data() ?? defaultData;
+    data = data.filter((entry) => entry.type === LeaderboardEntry.TypeEnum.User).sort(sortByScore);
```

Comment:
I think this logic could be moved to the server side. Sorting and filtering out the bot users

Reply:
I was under the assumption that in one of the next versions we want to implement custom table filters / sorting for the user. So that for example a user is able to display BOT-accounts or sort after a different criterium.
Then we need to have filtering/sorting logic in the client anyways I guess?

Reply:
I don't think that code rabbit is interesting to see on the leaderboard we don't need a filter function for just hiding that. I think we can just generally exclude it.

Sorting does also not make sense to change for a leaderboard, or at least is not essential at the moment. 

The next step would be filtering in regard to time frame IMO, so you can see the leaderboard over the last week, month, custom or so.

Reply:
I personally thought about either timeframe or repository filters (as soon as we start supporting multiple repos).

But you are right actually, it indeed might make more sense to move the logic to the server. Will change that.

Reply:
True repository filters also, we can just have request parameters for the filtering and let the server take care of it. Angular/react query works really well with that.

Reply:
Done in 7472b20.